docs/user/sampling.md

```diff
@@ -0,0 +1,67 @@
+# Sampling
+
+`LocalRunner` gets trajectories through Sampling to train the policy. In
+Garage, [`LocalRunner`](https://github.com/rlworkgroup/garage/blob/a767dcb95988abc14165a2a320ee39932a1c85ca/src/garage/experiment/local_runner.py)
+uses [`Sampler`](https://github.com/rlworkgroup/garage/blob/a767dcb95988abc14165a2a320ee39932a1c85ca/src/garage/sampler/sampler.py)
+to perform sampling. `Sampler` manages [`Worker`](https://github.com/rlworkgroup/garage/blob/a767dcb95988abc14165a2a320ee39932a1c85ca/src/garage/sampler/worker.py#L5)s
+and assign specific tasks to them, such as performing rollouts and update
+agents and environments.
+
+## Sampler
+
+Sampler is responsible for assign sampling jobs to workers. Garage now has
+three types of `Sampler`s:
+
+- [`LocalSampler`](https://github.com/rlworkgroup/garage/blob/a767dcb95988abc14165a2a320ee39932a1c85ca/src/garage/sampler/local_sampler.py),
+the default sampler, which runs workers in the main process. With this sampler,
+all the sampling tasks will be run in the same thread.
+
+- [`MultiprocessingSampler`](https://github.com/rlworkgroup/garage/blob/master/src/garage/sampler/multiprocessing_sampler.py),
+the sampler use [multiprocessing](https://docs.python.org/3/library/multiprocessing.html)
+package to run workers in different processes. With multiprocessing, the
+sampler can avoid waiting as much as possible.
+
+- [`RaySampler`](https://github.com/rlworkgroup/garage/blob/master/src/garage/sampler/ray_sampler.py),
+the sampler use [ray](https://github.com/ray-project/ray) framework to run
+distributed workers. `RaySampler` can not only run workers in different CPUs,
+but also in different machines.
+
+## Worker
+
+Worker is the base unit to perform a rollout and update the agent and the
+environment. In paralleling samplers, a worker will typically run in one
+exclusive CPU. Garage currently have two kinds of workers:
```

Comment:
You should mention that these are the two workers that can be used for most Algorithms, but a few Algorithms (RL2 and PEARL) use custom workers specific to that Algorithm.